test/cpu_ops_test.cpp

```diff
@@ -1763,4 +1763,79 @@ TEST_CASE(reduce_sum_test12)
     EXPECT(results_vector == gold);
 }
 
+TEST_CASE(reduce_mean_test1)
```

Comment:
can we rename these tests to reduce_mean_axis_{1,2,...}

Reply:
I notice sum tests above have the same naming convention, but feel free to change them too

Reply:
can we also get a test for no axis provided? The default should be to reduce all dimensions if no axis is provided, both for onnx and tf.

Reply:
Added and add shape test to cover code change